ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivacyController.java

```diff
@@ -80,37 +81,41 @@ public PrivateTransactionHandler(
     this.privateMarkerTransactionFactory = privateMarkerTransactionFactory;
   }
 
-  public String sendToOrion(final PrivateTransaction privateTransaction) {
-    final SendRequest sendRequest = createSendRequest(privateTransaction);
-    final SendResponse sendResponse;
-
+  public SendTransactionResponse sendTransaction(final PrivateTransaction privateTransaction) {
     try {
       LOG.trace("Storing private transaction in enclave");
-      sendResponse = enclave.send(sendRequest);
-      return sendResponse.getKey();
+      final SendResponse sendResponse = sendRequest(privateTransaction);
+      final String enclaveKey = sendResponse.getKey();
+      if (privateTransaction.getPrivacyGroupId().isPresent()) {
```

Comment:
nit: can we raise a ticket to determine the group directly from the txn - rather than uploading to enclave first?